cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,673 @@
+        # we can use the CausalGraph.__eq__ method and then check for the metadata
+        are_equal = super().__eq__(other)
+        # no need to check the metadata if the graphs are not equal
+        if not are_equal:
+            return False
+
+        # now check the metadata timedelta in the nodes
+        for node in self.get_nodes():
+            other_node = other.get_node(node.identifier)
+            # check the variable name
+            node_metadata_variable = node.get_metadata()[self._meta_variable_name]
+            other_node_metadata_td = other_node.get_metadata()[self._meta_variable_name]
+
+            if node_metadata_variable != other_node_metadata_td:
+                return False
+
+            # check the time delta
+            node_metadata_td = node.get_metadata()[self._meta_time_name]
+            other_node_metadata_td = other_node.get_metadata()[self._meta_time_name]
+
+            if node_metadata_td != other_node_metadata_td:
+                return False
+
+        return True
+
+    def copy(self, include_meta: bool = True) -> TimeSeriesCausalGraph:
+        """Return a copy of the graph."""
+        graph = super().copy(include_meta=include_meta)
+        # cast the graph to TimeSeriesCausalGraph to have the correct metadata
+        return TimeSeriesCausalGraph.from_causal_graph(graph)
+
+    def get_minimal_graph(self) -> TimeSeriesCausalGraph:
+        """
+        Return a minimal graph.
+
+        The minimal graph is the graph with the minimal number of edges that is equivalent to the original graph.
+        In other words, it is a graph that has no edges whose destination is not time delta 0.
+        """
+        minimal_cg = TimeSeriesCausalGraph()
+
+        # first step
+        # go through all the nodes and add them to the minimal graph if they have time delta 0
+        for node in self.get_nodes():
+            if node.get_metadata()[self._meta_time_name] == 0:
+                # add if node is not already in the graph
+                if not minimal_cg.node_exists(node.identifier):
+                    minimal_cg.add_node(node=node)
+
+        # second step: add nodes that are connected to the nodes in the minimal graph
+        # go through all the nodes in the minimal graph and add the nodes that are entering them
+        for node in minimal_cg.get_nodes():
+            for edge in self.get_edges():
+                # could also check just edge.destination == node (avoiding the metadata check)
+                if edge.destination.identifier == node.identifier:
+                    # add the node if it is not already in the graph
+                    if not minimal_cg.node_exists(edge.source):
+                        minimal_cg.add_node(node=self.get_node(edge.source))
+                    minimal_cg.add_edge(edge=edge)
+
+        return minimal_cg
+
+    def is_minimal_graph(self) -> bool:
+        """Return True if the graph is minimal."""
```

Comment:
again not clear here what the minimal graph is. can point to another doc string here if you just want to explain in one place